core/mybatis-generator-core/src/main/java/org/mybatis/generator/api/Plugin.java

```diff
@@ -2044,4 +2044,30 @@ default boolean clientUpdateByPrimaryKeyMethodGenerated(KotlinFunction kotlinFun
     default boolean shouldGenerate(IntrospectedTable introspectedTable) {
         return true;
     }
+
+    /**
+     * The motivation for adding this method can be found in
+     * https://github.com/mybatis/generator/issues/1116
+     *
+     * This method is called when the updateByPrimaryKey method
+     * has been generated in the client interface.
```

Comment:
Please add a comment stating this method is only called in the dynamic SQL runtime.